Disher/View Controllers/SaveViewController.m

```diff
@@ -43,17 +43,33 @@ - (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForR
 }
 
 - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
-    //TODO: CHECK FOR ALREADY EXISTING ENTRY
     Recipe *selectedRecipe = self.passedRecipe;
-    [selectedRecipe saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+    PFQuery *query = [PFQuery queryWithClassName:@"Recipe"];
+    [query whereKey:@"recipeID" equalTo:selectedRecipe.recipeID];
+    [query whereKey:@"dishName" equalTo:selectedRecipe.dishName];
+    [query whereKey:@"source" equalTo:selectedRecipe.source];
+
+    Recipe *foundObject = [query getFirstObject];
+    if (foundObject) {
         List *selectedList = self.lists[indexPath.row];
-        [selectedList addUniqueObject:selectedRecipe.objectId forKey:@"recipes"];
+        [selectedList addUniqueObject:foundObject.objectId forKey:@"recipes"];
         [selectedList saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
             if (!error) {
                 [self dismissViewControllerAnimated:YES completion:nil];
             }
         }];
-    }];
+    }
+    else {
+        [selectedRecipe saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+            List *selectedList = self.lists[indexPath.row];
+            [selectedList addUniqueObject:selectedRecipe.objectId forKey:@"recipes"];
+            [selectedList saveInBackgroundWithBlock:^(BOOL succeeded, NSError * _Nullable error) {
+                if (!error) {
+                    [self dismissViewControllerAnimated:YES completion:nil];
+                }
+            }];
+        }];
```

Comment:
this logic seems duplicated from above. Can we separate it into a helper function and pass in the object ID?